msger_mosaic_bridge/src/mosaic_adapter.cpp

```diff
@@ -199,27 +204,54 @@ void MosaicAdapter::on_siren_and_light_status_recieved_handler(bool siren_active
 
 void MosaicAdapter::on_traffic_event_received_handler(float up_track, float down_track, float minimum_gap, float advisory_speed)
 {
-    auto request = std::make_shared<carma_msgs::srv::SetTrafficEvent::Request>();
-    request->up_track = up_track;
-    request->down_track = down_track;
-    request->minimum_gap = minimum_gap;
-    request->advisory_speed = advisory_speed;
-
-    auto future_result = traffic_event_client_->async_send_request(request, 
-        [this](rclcpp::Client<carma_msgs::srv::SetTrafficEvent>::SharedFuture future) {
-            // Handle the response from the service
-            auto response = future.get();
-            if (response->success)
-            {
-                RCLCPP_INFO(this->get_logger(), "Traffic event set successfully.");
-            }
-            else
-            {
-                RCLCPP_ERROR(this->get_logger(), "Failed to set traffic event.");
-            }
-    });
+    // Check if all parameters are zero
+    if (up_track == 0 && down_track == 0 && minimum_gap == 0 && advisory_speed == 0)
```

Comment:
Comparing floats/doubles using `==` is not usually advisable, but may work in this case. Floating-point precision makes `==` frequently unreliable in ways that are tricky to understand and catch. Usually a comparison is made to say "a == b within some small error factor `epsilon`". I would simply caution to pay attention during testing in case this causes issues, this may be okay since we're _specifically_ passing in integer-literal 0s for first call we make.

Just wanted to make a note.